x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/planning/AssignmentPlan.java

```diff
@@ -315,6 +314,14 @@ private Quality computeQuality() {
         return new Quality(isSatisfyingPreviousAssignments, weighedAllocationsScore, memoryScore);
     }
 
+    public AssignmentPlan withDeploymentsWithZeroAllocations(Collection<Deployment> deploymentsWithZeroAllocations) {
+        Map<Deployment, Map<Node, Integer>> newAssignments = new HashMap<>(assignments);
+        for (Deployment deployment : deploymentsWithZeroAllocations) {
+            newAssignments.put(deployment, Collections.emptyMap());
+        }
+        return new AssignmentPlan(newAssignments, remainingNodeMemory, remainingNodeCores, remainingModelAllocations);
```

Comment:
```suggestion
    public AssignmentPlan withDeploymentsWithZeroAllocations(Collection<Deployment> deploymentsWithZeroAllocations) {
        Map<Deployment, Map<Node, Integer>> newAssignments = new HashMap<>(assignments);
        Map<Deployment, Integer> newRemainingModelAllocations = new HashMap<>(remainingModelAllocations);
        for (Deployment deployment : deploymentsWithZeroAllocations) {
            newAssignments.put(deployment, Collections.emptyMap());
            newRemainingModelAllocations.put(deployment, 0);
        }
        return new AssignmentPlan(newAssignments, remainingNodeMemory, remainingNodeCores, newRemainingModelAllocations);
```

I would add the zero-allocation deployments to the `remainingModelAllocations` map for consistency. Right now it work fine even without this change, since `satisfiesAllocations()` handles missing deployments as if they have 0 remaining allocations, but if we rely on having all deployments in `remainingModelAllocations` elsewhere in the code, it may introduce a subtle bug.